### ISSUE TITLE
Improve handling of  <see cref="..." /> and <paramref name="..." /> tags in .NET XML doc comments

### ISSUE THREAD

**Issue by danroth27** [rtfd/sphinx-autoapi#56]
The `<see cref"..." />` tags in .NET XML doc comments frequently show up the generated rst or sometimes get completely removed. In some cases `<paramref name="..." />` tags are also not handled. Take this example:

```c#
         /// <summary> 
         /// Get the security stamp for the specified <paramref name="user" />. 
         /// </summary> 
         /// <param name="user">The user whose security stamp should be set.</param> 
         /// <param name="cancellationToken">The <see cref="CancellationToken"/> used to propagate notifications that the operation should be canceled.</param> 
         /// <returns>The <see cref="Task"/> that represents the asynchronous operation, containing the security stamp for the specified <paramref name="user"/>.</returns> 
         public virtual Task<string> GetSecurityStampAsync(TUser user, CancellationToken cancellationToken = default(CancellationToken))
         { 
             return Task.FromResult("abc"); 
         } 
```

The generated rst looks like this:

```
    .. dn:method:: ClassLibrary3.UserStore<TUser, TRole, TContext, TKey>.GetSecurityStampAsync(TUser, System.Threading.CancellationToken)
    
        
    
        Get the security stamp for the specified ``user``.
    
        
        
        
        :param user: The user whose security stamp should be set.
        
        :type user: {TUser}
        
        
        :param cancellationToken: The  used to propagate notifications that the operation should be canceled.
        
        :type cancellationToken: System.Threading.CancellationToken
        :rtype: System.Threading.Tasks.Task{System.String}
        :return: The <see cref="T:System.Threading.Tasks.Task" /> that represents the asynchronous operation, containing the security stamp for the specified <paramref name="user" />.
```
A few things to note:
- The paramref is handled in the summary, but not in the return value description.
- The `<see cref="T:System.Threading.Tasks.Task" />` element is left in the return value description instead of being handled. 
- The `<see cref="CancellationToken"/>` element is completely removed from the description for the user parameter without leaving any text at all.

**Comment by danroth27** [comment]
Adding YAML metadata output from docfx.


[yaml.zip](https://github.com/rtfd/sphinx-autoapi/files/115214/yaml.zip)

**Comment by agjohnson** [comment]
These look mostly to be parsing errors. Besides the last issue, with `CancellationToken`, which seems to be missing the tag in the description in YAML output:

```yaml
    - id: cancellationToken
      type: System.Threading.CancellationToken
      description: The  used to propagate notifications that the operation should be canceled.
```

**Comment by agjohnson** [comment]
The return type should have been parsed out correctly, but the underlying fields weren't updated correctly when they changed last. Also looks like there were issues with `paramref` parsing. I've addressed these in #64, along with adding some tests.
